docs/pages/docs/guides/schema-extension.md

```diff
@@ -1,6 +1,125 @@
+It then returns the Post that is updated which has a GraphQL type of `Post` that is passed in from the `base` schema (`base.object('Post')`).
+
+```ts
+import { graphql, config } from '@keystone/core';
+import { Context } from '.keystone/types';
+
+export default config({
+  {/* ... */},
+  extendGraphqlSchema: graphql.extend(base => {
+    return {
+      mutation: {
+        publishPost: graphql.field({
+          type: base.object('Post'),
+          args: { id: graphql.arg({ type: graphql.nonNull(graphql.ID) }) },
+          resolve(source, { id }, context:Context) {
+            return context.db.Post.updateOne({
+              where: { id },
+              data: { status: 'published', publishDate: new Date().toISOString() },
+            });
+          },
+        }),
+      },
+    };
+  }),
+});
+```
+
+{% hint kind="tip" %}
+Note `context.db` is used in the resolver, this ensures the correct Internal object with the correct type is returned to GraphQL.
+{% /hint %}
+
+A full example project using `graphql-ts` can be found in [examples/extend-graphql-schema-graphql-ts](https://github.com/keystonejs/keystone/tree/main/examples/extend-graphql-schema-graphql-ts) on the Keystone GitHub repo.
+
+## Using Third-Party Tools
+
+As `extendGraphqlSchema` expects a function that returns a valid GraphQL schema you can also use third-party GraphQL schema tools to help generate or merge schemas.
+
+### GraphQL-Tools Merge Schemas
+
+[GraphQL Tools](https://www.graphql-tools.com/) `mergeSchemas` is a third-party package that can help with schema merging and adding custom resolvers and types, and then return an updated GraphQL schema to Keystone.
+
+Start by installing `@graphql-tools/schema`
+
+```bash
+yarn add @graphql-tools/schema
+```
+
+Then import into your Keystone configuration
+
+```ts
+import { mergeSchemas } from '@graphql-tools/schema';
+```
+
+You can then write custom `resolvers` and `typeDefs` to merge with your Keystone `schema`. For example, to add the same custom mutation as above (`publishPost`), you would add the `typeDefs` for `publishPost` and then the `resolvers.Mutation.publishPost`
+which performs the same function as above.
+
+```ts
+export default config({
+  {/* ... */},
+  extendGraphqlSchema: schema =>
+    mergeSchemas({
+      schemas: [schema],
+      typeDefs: `
+      type Mutation {
+        publishPost(id: ID!): Post
+      `,
+      resolvers: {
+        Mutation: {
+          publishPost: (root, { id }, context) => {
+            return context.db.Post.updateOne({
+              where: { id },
+              data: { status: 'published', publishDate: new Date().toUTCString() },
+            });
+          },
+        },
+      },
+    }),
+});
+```
+
+{% hint kind="tip" %}
+Note - Before version `3.0.0` of Keystone-6, `@graphql-tools/schema` was exported from `@keystone/core` as `graphQLSchemaExtension` this was removed in favor of using the tool directly if required
+{% /hint %}
```

Comment:
```suggestion

{% if $nextRelease %}
{% hint kind="tip" %}
Note - Before version `3.0.0` of `@keystone-6/core`, `@graphql-tools/schema` was exported from `@keystone/core` as `graphQLSchemaExtension` this was removed in favor of using the tool directly if required
{% /hint %}
{% /if %}
```

(if you'd like to add an else to say something like "it's gonna be removed soon", that might be useful but I don't mind either way)

Reply:
Thanks @mitchellhamilton ... went with the `else` option.